pkg/controller/direct/colab/runtimetemplate_controller.go

```diff
@@ -0,0 +1,284 @@
+				return nil, err
+			}
+			return &runtimeTemplateAdapter{
+				gcpClient: gcpClient,
+				id:        id,
+			}, nil
+		}
+	}
+	return nil, nil
+}
+
+type runtimeTemplateAdapter struct {
+	gcpClient *gcp.NotebookClient
+	id        *krm.NotebookRuntimeTemplateIdentity
+	desired   *pb.NotebookRuntimeTemplate
+	actual    *pb.NotebookRuntimeTemplate
+}
+
+var _ directbase.Adapter = &runtimeTemplateAdapter{}
+
+func (a *runtimeTemplateAdapter) Find(ctx context.Context) (bool, error) {
+	log := klog.FromContext(ctx)
+	log.V(2).Info("getting colabruntimetemplate", "name", a.id)
+
+	req := &pb.GetNotebookRuntimeTemplateRequest{Name: a.id.String()}
+	actual, err := a.gcpClient.GetNotebookRuntimeTemplate(ctx, req)
+	if err != nil {
+		if direct.IsNotFound(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("getting colabruntimetemplate %q from gcp: %w", a.id.String(), err)
+	}
+
+	a.actual = actual
+	return true, nil
+}
+
+func (a *runtimeTemplateAdapter) Create(ctx context.Context, createOp *directbase.CreateOperation) error {
+	log := klog.FromContext(ctx)
+	log.V(2).Info("creating colabruntimetemplate", "name", a.id)
+
+	desired := direct.ProtoClone(a.desired)
+	desired.Name = a.id.String()
+
+	req := &pb.CreateNotebookRuntimeTemplateRequest{
+		Parent:                    a.id.Parent().String(),
+		NotebookRuntimeTemplate:   desired,
+		NotebookRuntimeTemplateId: a.id.ID(),
+	}
+	op, err := a.gcpClient.CreateNotebookRuntimeTemplate(ctx, req)
+	if err != nil {
+		return fmt.Errorf("creating colabruntimetemplate %s: %w", a.id.String(), err)
+	}
+	created, err := op.Wait(ctx)
+	if err != nil {
+		return fmt.Errorf("colabruntimetemplate %s waiting creation: %w", a.id, err)
+	}
+	log.V(2).Info("successfully created colabruntimetemplate in gcp", "name", a.id)
+
+	status := &krm.ColabRuntimeTemplateStatus{}
+	mapCtx := &direct.MapContext{}
+	status.ObservedState = ColabRuntimeTemplateObservedState_FromProto(mapCtx, created)
+	if mapCtx.Err() != nil {
+		return mapCtx.Err()
+	}
+	status.ExternalRef = direct.PtrTo(a.id.String())
+	return createOp.UpdateStatus(ctx, status, nil)
+}
+
+func (a *runtimeTemplateAdapter) Update(ctx context.Context, updateOp *directbase.UpdateOperation) error {
+	log := klog.FromContext(ctx)
+	log.V(2).Info("updating colabruntimetemplate", "name", a.id)
+
+	desired := direct.ProtoClone(a.desired)
+	desired.Name = a.id.String()
+
+	// TODO(user): Update the field if applicable.
+	updateMask := &fieldmaskpb.FieldMask{}
+	if !reflect.DeepEqual(a.desired.DisplayName, a.actual.DisplayName) {
+		updateMask.Paths = append(updateMask.Paths, "display_name")
+	}
+	if !reflect.DeepEqual(a.desired.Description, a.actual.Description) {
+		updateMask.Paths = append(updateMask.Paths, "description")
+	}
+	if !reflect.DeepEqual(a.desired.Etag, a.actual.Etag) {
+		updateMask.Paths = append(updateMask.Paths, "etag")
+	}
+	if !reflect.DeepEqual(a.desired.Labels, a.actual.Labels) {
+		updateMask.Paths = append(updateMask.Paths, "labels")
+	}
```

Comment:
Is there any reasons to not use CompreProtoMessage?

Reply:
No. It was generated.

Updated.